src/ansys/templates/python/pyace_fastapi/{{cookiecutter.__project_name_slug}}/Docker.md

```diff
@@ -0,0 +1,57 @@
+## Containerized deployment
+
+### Usage
+
+#### Build
+
+```sh
+$ docker build \
+    -t $USER/{{cookiecutter.__project_name_slug}} \
+    .
+```
+
+```ps1
+> docker build `
+    -t $env:UserName/{{cookiecutter.__project_name_slug}} `
+    .
+```
```

Comment:
If you change the Dockerfile to docker/Dockerfile, you would just need to specify the Dockerfile location with ``-f docker/Dockerfile`` and keep the root directory as the main directory for the build (so that it detects all other files). I am guessing you moved the docker/* files to the root level because you weren't able to access the rest of the files, but there are ways to do it.

See for other examples building docker images with a `docker` folder. The docs show the commands to build the images while accessing all repository files
* https://apieigen.docs.ansys.com/docker/index.html
* https://github.com/ansys/api-eigen-example

Reply:
More specifically, if you want to build a Docker image from a specific Dockerfile (at some random path), with all the files of the root level directory, you just have to run the following command at the root level.

```bash
docker image build -f {my-Dockerfile} .
```

Reply:
I was aware of the `-f` flag. This said, from my point of view, if you ask for `docker support` you want it to be a first class citizen within your project. As we already have `docker-compose.yml`, and because we need `.dockerignore` at the root, I don't see the point of not having `Dockerfile` as well. Remains the documentation file which could be relocated somewhere within `doc` folder (I should confess I am not aware of how your `sphinx` stuff works)

I can revert my changes if you really want, but I don't consider hiding `Dockerfile` within a `docker` folder a proper guidelines. It could make sense at some point when project scales - ie provide multiple `Dockerfile` - but not with a single one.

Reply:
I honestly meant placing all docker-related files under the docker folder (this includes `docker-compose.yml`, `Dockerfile` and `.dockerignore`), not just the `Dockerfile` . Unless your project is exclusively a docker project it is a good practice to keep all docker related files in a docker folder. It is not "hiding" a Dockerfile what I suggest, it is just placing all docker-related files in one same directory, instead of having them at root level. The same way we do with requirement files (i.e. `requirements`) and documentation (i.e. `doc`).

Anyway, it is a suggestion from my side. Feel free to adopt the solution you prefer.

Reply:
1. `.dockerignore` needs to be root to play its role. We have no choice.
2. `docker-compose.yml` & `Dockerfile` act as main entry points, so should be at root as well. Remember, docker support is explicitly asked by user during project creation, so he expects this to be clearly materialized. And, as it is located at root, it pushes user to use it, instead of the plain old `py ./src/server.py`. As there are some many ways to fail locally, like forgetting to create `venv`, forgetting to active `venv`, forgetting to freeze `requirements` after `pip installing` locally, ... it should also be pushed as the de facto way of running the program on a daily basis. Once you taste it, you can't move backward (like wanna know if your program upgrade from `3.7` to `3.10`, change a line in the `Dockerfile` and relaunch, or wanna know if your program is really OS agnostic, ...)
3. When project complexifies/scales, we could end up with multiple `Dockerfile` and `docker-compose.yml`. There is multiple schools, one could prefer to tidy up with a `Dockerfile` located in `tests`, one other in `src`. Some may even prefer to gather them in a `release` or `build` folder to better express the semantic purpose instead of the tech. As we do for others stuff, we put `*.py` in a `src` folder, not within a `python` one.
4. Regarding `Docker.md`, it seems to me it should be easily accessible as it supports new comers, and I think putting it down below `doc` hierarchy may defeat this purpose. Except if we fill `README.md` accordingly, providing hyperlink navigation or integrating `Docker.md` content into it. This said, regarding `requirements`, I do agree it can make sense to create a subfolder to gather them all, but we should also probably enrich `README` sections accordingly. Every `python` project I was exposed to is used to provide gently reminder about `virtualenv` as well, so we probably should as well

Once again, I expressed my view and the way I am used to regarding containerization and project organization. We should agree upon rules that make sense and everyone is comfortable with.

Reply:
After looking at projects like [PyMAPDL], [PyFluent], and considering that originally the files were collected under a `docker/` directory, I think we should stick with the rest of projects. It is just a matter of unifying projects structure across PyAnsys and Python projects developed by Ansys.

As @RobPasMue said, it makes sense to have all docker-related files within the same folder.

Pinging here @asbfall, as he implemented the templates affected by this pull request.

Reply:
As you want, I'll rollback changes on that matter.
This said, I doubt promoting projects that live w/o `.dockerignore` as `docker champion` makes a lot of sense...